core/multiproc/inc/TProcPool.h

```diff
@@ -35,8 +32,13 @@
 #include <algorithm> //std::generate
 #include <functional> //std::reference_wrapper
 #include <iostream>
+#include "TPool.h"
 
-class TProcPool : private TMPClient {
+template< class F, class... T>
+ using noReferenceCond = typename std::enable_if<"Function can't return a reference" && !(std::is_reference<typename std::result_of<F(T...)>::type>::value)>::type;
```

Comment:
Does this using stating needs to be in the global state?
It does not follow the name conventions.